src/aws/lambda-encode/src/encodeVideo.js

```diff
@@ -10,6 +15,48 @@ module.exports = async (objectKey, sourceBucket) => {
   const envType = process.env.ENV_TYPE;
   const destinationBucket = process.env.S3_DESTINATION_BUCKET;
 
+  const signedUrl = await s3.getSignedUrlPromise(
+    'getObject',
+    {
+      Bucket: sourceBucket,
+      Expires: 1200,
+      Key: objectKey
+    }
+  );
+
+  const { stdout, stderr } = await execFile('mediainfo', [
+    '--full',
+    '--output=JSON',
+    signedUrl
+  ]);
+  
+  // 29.970 fps
+  const framerateSettings = {
+    FramerateDenominator: 1001,
+    FramerateNumerator: 30000
+  }
+
+  if (stdout) {
+    const mediainfos = JSON.parse(stdout);
+
+    const videoInfo = mediainfos.media.track.find(track => track["@type"] === "Video")
+
+    switch(videoInfo.FrameRate) {
```

Comment:
Have you tried always calculating it? In pseudo code something like:

framerateSettings.FramerateDenominator = 1000 * round(videoInfo.FrameRate) / round(videoInfo.FrameRate * 1000)
framerateSettings.FramerateNumerator = 1000 * round(videoInfo.FrameRate)